plugins/experimental/cachekey/plugin.cc

```diff
@@ -91,8 +91,26 @@ TSRemapDoRemap(void *instance, TSHttpTxn txn, TSRemapRequestInfo *rri)
   Configs *config = (Configs *)instance;
 
   if (nullptr != config) {
+    TSMBuffer bufp;
+    TSMLoc urlLoc;
+    TSMLoc hdrLoc;
+
+    /* Get the URI and header to base the cachekey on.
+     * @TODO it might make sense to add more supported URI types */
+    if (PRISTINE == config->getUriType()) {
+      if (TS_SUCCESS != TSHttpTxnPristineUrlGet(txn, &bufp, &urlLoc)) {
+        /* Failing here is unlikely. No action seems the only reasonable thing to do from within this plug-in */
+        CacheKeyError("failed to get pristine URI handle");
+        return TSREMAP_NO_REMAP;
```

Comment:
I'm thinking we maybe should try to error out the entire request in this scenario? We clearly can't make the cache key that we want. I'm ok with this, if @SolidWallOfCode and @bryancall are (since this seems to be their use case).

Reply:
It's likely this error case can never happen either, not sure?

Reply:
50x sounds reasonable to me, but currently the `cachekey` follows "no action on TS API failure and error in the log" everywhere so I thought I would keep it consistent, noticed that `cacheurl` plugin handled this supposedly rare case (TS API failure) similarly.

Will need to make bigger change to `cachekey` plugin (think more + different PR) to error and return 50x (+ may be `TSREMAP_NO_REMAP_STOP` ?)